src/services/combinePath.ts

```diff
@@ -22,5 +21,11 @@ export function combinePath(parentPath: Path, childPath: Path): Path {
     throw new DuplicateParamsError(key)
   }
 
-  return path(`${parentPath.path}${childPath.path}`, { ...parentPath.params, ...childPath.params })
+  const newPathString = `${parentPath.path}${childPath.path}`
+
+  return {
+    path: newPathString,
+    params: { ...parentPath.params, ...childPath.params },
+    toString: () => newPathString,
+  }
```

Comment:
when combining parent path and child path, we used to use `path` again. The problem with this approach is the parent `path` has already been called, (child might have been too if done so by user). When calling `path`, the params are expected to be without `?`. This allows developers to call path as expected `path('/something/[?id]', {id: Number})` note the `id` in params doesn't expect `?`. Without this change, the nested `path` call would lose `Number` type because `path.params`  after being called initially would be `{ '?id': Number }`. 

Alternatively, if you don't like this we can have path and query both look for with and without `?` since we don't allow for params of the same name, even if one has `?` and the other doesn't.